.github/workflows/maven.yaml

```diff
@@ -82,3 +89,55 @@ jobs:
         with:
           name: container-logs
           path: "**/container-logs/**/*.log"
+
+  older_kafka_matrix:
+    runs-on: ubuntu-latest
+    needs: build
+    strategy:
+      matrix:
+        version: [3.8.0]
+    steps:
+      - name: 'Check out repository'
+        uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+          fetch-depth: 0
+
+      - name: 'Set up Java'
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'zulu'
+          cache: maven
+
+      - name: 'Cache Maven packages'
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: 'Install junit extension'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: mvn -B install -DskipTests=true
+
+      - name: 'Run select junit tests (Kafka in-VM)'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        # Note - the extension won't necessarily compile against the earlier versions of Kafka.
```

Comment:
Maven doesn't make this easy.  Ideally you'd have project produce a test jar, then you'd point surefire at that artefact.  However, Maven doesn't allow such things.  I've banged my head for hours here before.

Instead, I'm running the tests against the installed artefacts.   The Junit5 test code needs to be compatible with older Kafka clients, otherwise we'd fail to compile.

Reply:
I have seen hacks producing `-test` jars though I can't remember which plugin was used to generate the additional artefact.

Reply:
IIRC producing test jars is not too hard.  The diffcult was actually getting surefire to execute the tests from the JAR, rather than from compiled source files on a filesystem.